unlock-app/src/components/interface/locks/Settings/forms/UpdateHooksForm.tsx

```diff
@@ -27,7 +29,50 @@ interface FormProps {
   keyGrant?: string
 }
 
-type FormPropsKeys = keyof FormProps
+type FormPropsKey = keyof FormProps
+interface HookValueProps {
+  label: string
+  version: number
+  hookName: HookName
+}
+
+const HookMapping: Record<FormPropsKey, HookValueProps> = {
+  keyPurchase: {
+    label: 'Key purchase hook',
+    version: 7,
```

Comment:
```suggestion
    fromPublicLockVersion: 7,
```

This will be clearer ^^